packages/commercetools/theme/components/AppTopBar.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <div class="sf-topbar">
+    <div class="sf-topbar__content">
+      <div class="sf-topbar__content__item"></div>
+      <div class="sf-topbar__content__item">Download our application. <a href="#">Find out more.</a></div>
+      <div class="sf-topbar__content__item selector-container">
+        <SfSelect
+          v-model="selectedLanguage"
+          @change="onLanguageSelect"
+      >
+        <SfSelectOption
+          v-for="language in languages"
+          :key="language.value"
+          :value="language.value"
+        >
+          {{language.label}}
+        </SfSelectOption>
+      </SfSelect>
+      <SfSelect
+          v-model="selectedCountry"
+          @change="onCountrySelect"
+      >
+        <SfSelectOption
+          v-for="country in countries"
+          :key="country.value"
+          :value="country.value"
+        >
+          {{country.label}}
+        </SfSelectOption>
+      </SfSelect>
+      </div>
+      
+    </div>
+  </div>
+</template>
+
+<script>
+import { useLocale } from '@vue-storefront/commercetools-composables'
+import { reactive } from '@vue/composition-api'
+import { SfSelect } from "@storefront-ui/vue";
+
+export default {
+  setup(props, { isServer }) {
+    const { getLanguage, setLanguage, getCountry, setCountry, getCountries } = useLocale()
+    const countries = reactive([])
+
+    let selectedLanguage = '';
+    let selectedCountry = '';
+
+    if (!isServer) {
+      selectedLanguage = (localStorage.getItem('language') || 'en').toLowerCase()
```

Comment:
that will cause a mismatch (probably) - imagine that on the server you have English but on the client-side you have saved German. In that case, a response that comes from the server is in English but every next request is in German (because you just changed lang)

Reply:
Yes, that's what I've mentioned in previous review. Once we'll have ready SSR, we'll move local-storage for cookies so it will be supported on SSR and CSR side as well